### DISCUSSION TITLE
1.3.3

### DISCUSSION THREAD

**Discussion by ilyhalight** [ilyhalight/voice-over-translation#213] [Обновления]
### Список изменений
- Возвращен отступ кнопки в 5rem
- Добавлены новые переводы в meta скрипта
- Элементы меню, теперь, правильно строятся, если имеют слишком короткие названия
- Теперь, в одной из ошибок, связанных с обновлением базы данных пишется домен текущего сайта, а не статический "youtube.com"
- availableFromLangs и availableToLangs заменены одной переменной availableLangs
- Создание списков доступных языков для перевода и языков видео приведены к единому стилю
- Добавлен метод строки format, который служит для форматирования строк перевода
- Расширение было полностью переведено на 7 доступных языков (Русский, Английский, Немецкий, Китайский, Испанский, Итальянский, Французский). Язык меню на данный момент выбирается на основе языка браузера - [ilyhalight/voice-over-translation#212](https://github.com/ilyhalight/voice-over-translation/pull/212)

## What's Changed
* Перевод меню (неполный) by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/212

**Full Changelog**: https://github.com/ilyhalight/voice-over-translation/compare/1.3.2...1.3.3

<hr /><em>This discussion was created from the release <a href='https://github.com/ilyhalight/voice-over-translation/releases/tag/1.3.3'>1.3.3</a>.</em>

**Comment by FiLinX** [comment]
Сейчас столкнулся со странностью  -  вот на этом ролике https://youtu.be/Qyn64b4LNJ0 
Во первых, удивило, что на нём пришлось ждать перевода  - ну, обычно на известных каналах у яндыкса уже всё переведено и перевод приходит мгновенно из его кешей, а тут вдруг задержка..
Но далее, дождавшись "Включено", запускаю с паузы видос  -  а перевод идёт с инглиша на инглишь)) хотя в панельке то всё верно стоит англ-рус 
Я захотел как-то восстановить справедливость, сменил вместо рус на немец, вновь пошло "ожидайте", и когда вновь увидел "включено" запустил видос с паузы - услышал нормальный перевод на немецком.
Таак, а теперь вот настала очередь вернуть англ-рус - открываю панельку, а там блин так и прописано англ-рус, хотя перевод то идёт на немецком.. ыы)
Пришлось снова выбрать другой язык - испано, появилось ожидайте, я снова в панельку и вот теперь уже вместо испано выставил рус, дождался, включил с паузы - и наконец услышал понимаемую речь! ))

Я хз чё это было, но возможно повлияло, что я вчера ещё на другом ролике тоже тыркал в справа на англ в русском ролике, но перевода не дождался потому что у нас свет вырубили и всё потухло), а потом через несколько часов уже, я накатил обновлением 1.3.3 на бывшую 1.3.2, но до сегодня так перевод и не запускал..
Видимо назначение языка перевода где-то запоминается, но в панельке всегда выставляет англ-рус автоматом, хотя на самом деле перевод то будет получен на тот язык на который до этого кликали..

**Comment by FiLinX** [comment]
хм, вот опять такое произошло  -  перевод пришёл с англ на англ, хотя и в панели [стоит англ-рус](https://i.ibb.co/bF84SVG/screenshot-2023-07-12-21-27-50-001.png) и до этого я только такое и юзал..
то есть происходит какой-то сбой авто-определялки языка видео
плохо то  -  что вот выставить то принудительно чтоб англ-рус получить  -  слишком геморройно через эти переназначения на другие языки

возможно ли - вообще исключить весь функционал определялки языка видео? ну чтоб он всегда по дефолту только на англ-рус запускался (даже на русских видосах) - ну то есть вообще выключить эту автоопределялку

**Comment by NullVerdict** [comment]
> хм, вот опять такое произошло - перевод пришёл с англ на англ, хотя и в панели [стоит англ-рус](https://i.ibb.co/bF84SVG/screenshot-2023-07-12-21-27-50-001.png) и до этого я только такое и юзал.. то есть происходит какой-то сбой авто-определялки языка видео плохо то - что вот выставить то принудительно чтоб англ-рус получить - слишком геморройно через эти переназначения на другие языки
> 
> возможно ли - вообще исключить весь функционал определялки языка видео? ну чтоб он всегда по дефолту только на англ-рус запускался (даже на русских видосах) - ну то есть вообще выключить эту автоопределялку
Попробуй мой [билд](https://github.com/SashaXser/voice-over-translation/tree/master/dist), и отпиши об изменениях

**Reply by FiLinX** [reply]
Отписываю:
Скрипт ставил заново, с удалением стоявшего и перезапуском темпермонки.
За сутки удалось посмотреть не более десятка роликов в переводе, и из них лишь два были не из кеша яндекса, а с ожиданием перевода - и это именно главный фактор как оказалось!
Вобщем вот только что на этом ролике https://youtu.be/o5cimXG53M0 было ожидание перевода, и как только он пришёл, запускаю - и слышу английский дубляж поверх английской речи ))
Что-то где-то запутано с определялками языка при запросе ещё не обрабатывавшегося яндыксом ролика..

**Comment by FiLinX** [comment]
ну, более-менее разобрался с этими англ-англ дубляжами..
во первых - они случаются на только-только выпеченных роликах, на которых ещё единицы просмотров.
а во вторых - ничего геморроить с переключениями языка ненадо  -  если при получении услышал англ-англ, то достаточно клацнуть F5 (перезагрузить вкладку) и вновь на "Перевести" - и сразу пойдёт на русском!
видимо от яндыкса англ дубляж сейчас первее приходит, на доли секунд раньше всех остальных

**Comment by FiLinX** [comment]
упс.. неет, свежесть ролика не причём..
сейчас вот опять словил англ-англ после ожидания на ролике вышедшем 11 месяцев назад и со 124 тысячами просмотров..
странный этот яндыкс)
